src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,102 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.exception.ProductException;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.repository.IProductCategoryRepository;
+import ro.msg.learning.shop.repository.IProductRepository;
+import ro.msg.learning.shop.repository.ISupplierRepository;
+import ro.msg.learning.shop.utils.Mapper;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class ProductService implements IProductService{
+    @Autowired
+    private IProductRepository productRepository;
+
+    @Autowired
+    private IProductCategoryRepository productCategoryRepository;
+
+    @Autowired
+    private ISupplierRepository supplierRepository;

```

Comment:
Same as in the other service. You should create a service for the Supplier where you can define the needed methods in order to use here the service and not the repository.